gsy_framework/data_classes.py

```diff
@@ -55,6 +55,7 @@ def __init__(self, id: str, creation_time: DateTime, price: float, energy: float
         self.energy = energy
         self.attributes = attributes
         self.requirements = requirements
+        self.type = self.__class__.__name__
```

Comment:
I remember a discussion about not using `type` as member of objects in order to not get confused with the build-in type. Can somebody else remember?

Reply:
I guess it must be OK to use the name `type` as an attribute since we are not overriding the `type` function in any scope of the program. Reading [this](https://stackoverflow.com/questions/30575543/is-using-type-as-an-attribute-name-a-bad-practice#:~:text=General%20rule%20is%3A%20don't,not%20really%20%22reserved%22) might also be useful.

Reply:
Further in the comment it says:
> In your case, since you're specifying type as a class attribute, it should be considered safe since it can only be accessed through its class (self.type or SomeProfile.type).

The reason I moved type to the __init__ method is that `obj.__dict__` will automatically return a dictionary of all the object's attributes and we almost always need `type` to be part of this dictionary; If we don't keep type in the object then we have to manually put the `type` key in the dictionary later in every place - which leads to code duplication:

- [Duplication1](https://github.com/gridsingularity/gsy-framework/pull/352/files#diff-3e3db6df6bdaeed8dd98e5be4a800b1e02fcbf5701f8486d1c577d72612a43a8L199)
- [Duplication2](https://github.com/gridsingularity/gsy-framework/pull/352/files#diff-f4f1059a8d7434f1f13437b37402c3fcfeec75b549221046672443b2ad219cf5L87)
- [Duplication3](https://github.com/gridsingularity/gsy-framework/pull/352/files#diff-f4f1059a8d7434f1f13437b37402c3fcfeec75b549221046672443b2ad219cf5L80)

I tried to avoid duplication of `self.__class__.__name__` by moving type into the `__init__` method.

Reply:
I see. Convinced.